testing/asserts.ts

```diff
@@ -864,3 +866,58 @@ export function unimplemented(msg?: string): never {
 export function unreachable(): never {
   throw new AssertionError("unreachable");
 }
+
+let snapshotFile: Record<string, unknown> | undefined = undefined;
+const updatedSnapshotFile: Record<string, unknown> = {};
+// const snapshotMap: Record<string, number> = {};
+
+export async function assertSnapshot(context: Deno.TestContext, actual: unknown) {
+  const name = getName(context);
+  // const count = getCount(name);
+
+  if (!snapshotFile) {
+    const snapshotPath = getSnapshotPath();
+    await ensureFile(snapshotPath);
+    const file = Deno.readTextFileSync(snapshotPath);
+    snapshotFile = file ? JSON.parse(file) : {};
+  }
+  const snapshot = snapshotFile?.[name];
+  if (!context.update) {
+    assertEquals(actual, snapshot);
```

Comment:
The current implementation will error on:

```ts
const actual = {
  fn: () => { /* ... */ }
}
```

As snapshots need to be represented as strings anyway, I would argue it makes more sense to use the `_format` helper here to convert the snapshot into a string. We could store the snapshot like this:

```json
{
  "Snapshot Test > babo > merong": [
    "{",
    "  b: 2,",
    "  c: 3,",
    "}"
  ]
}
```

This would enable us to catch a much wider range of errors in our snapshots. E.g. the following would error on `assertSnapshot`:

```ts
// First run
const actual = {
  testClass: class Test {}
}
// --snip--
assertSnapshot(test, actual)

// Second run
const actual = {
  testClass: class NotTest {}
}
// --snip--
assertSnapshot(test, actual)
```

But would not error (or worse, always error) if we only store the JSON representation.

Additionally, this representation enables us to use the `diff` and `buildMessage` to produce helpful output.

Reply:
I think [Deno.inspect()](https://doc.deno.land/deno/stable/~/Deno.inspect) can be a good choice for serialization method. What do you think? @bcheidemann

Reply:
Yeah this would definitely do the job. I would suggest to use the [`_format` helper](https://github.com/denoland/deno_std/blob/0d731bc7aeb6730d4c18da140c89c7d24651ba91/testing/asserts.ts#L31) which wraps `Deno.inspect`. This ensures the output will be consistent with other asserts 🙂